core/livepeernode.go

```diff
@@ -236,7 +236,7 @@ func (n *LivepeerNode) transcodeAndBroadcastSeg(seg *stream.HLSSegment, sig []by
 	start = time.Now()
 	for i, resultStrmID := range resultStrmIDs {
 		//Insert the transcoded segments into the streams (streams are already broadcasted to the network)
-		if tData[i] == nil {
+		if tData == nil || tData[i] == nil {
```

Comment:
Should we just return after L231 if `t.Transcode(seg.Data)` returns an error and not bother going into this loop since `tData` would always be `nil` in that case?